packages/at_client/lib/src/preference/at_client_preference.dart

```diff
@@ -100,6 +101,9 @@ class AtClientPreference {
   /// This instance variable is experimental, for now
   @experimental
   Version atProtocolEmitted = Version(1, 5, 0);
+
+  @experimental
+  AtClientParticulars atClientParticulars = AtClientParticulars.getInstance();
```

Comment:
Why a singleton?

Reply:
Since the appName, appVersion and platform are unique for the at_client instance, In future, if we want to access the data inside AtClientParticulars anywhere in at_client, it would be easy to call AtClientParticulars.getInstance(), then relying on "AtClientManager.atClient.getPreferences..".

Reply:
What about when switching atSigns? Using a singleton forces each client to have the same clientId - this is not what we want.

Singletons should never be created unless there is an architectural reason to have them. There's no such reason here.

Reply:
Got it Gary, will remove the singleton nature here.